tests_e2e/orchestrator/lib/agent_test_suite_combinator.py

```diff
@@ -150,89 +140,250 @@ def create_environment_list(self) -> List[Dict[str, Any]]:
+                                test_suite_info=test_suite_info)
+                        else:
+                            env = self.create_vm_environment(
+                                env_name=env_name,
+                                marketplace_image=marketplace_image,
+                                vhd=vhd,
+                                location=location,
+                                vm_size=vm_size,
+                                test_suite_info=test_suite_info)
+                        shared_environments[env_name] = env
+
+                    if test_suite_info.template != '':
+                        vm_tags = env.get("vm_tags")
+                        if vm_tags is not None:
+                            if "templates" not in vm_tags:
+                                vm_tags["templates"] = test_suite_info.template
+                            else:
+                                vm_tags["templates"] += "," + test_suite_info.template
 
         environments.extend(shared_environments.values())
 
         if len(environments) == 0:
             raise Exception("No VM images were found to execute the test suites.")
 
-        log: logging.Logger = logging.getLogger("lisa")
+        # Log a summary of each environment and the suites that will be executed on it
+        format_suites = lambda suites: ", ".join([s.name for s in suites])
+        summary = [f"{e['c_env_name']}: [{format_suites(e['c_test_suites'])}]" for e in environments]
+        summary.sort()
+        self._log.info("Executing tests on %d environments\n\n%s\n", len(environments), '\n'.join([f"\t{s}" for s in summary]))
+
         if len(skip_test_suites) > 0:
-            log.info("")
-            log.info("Test suites skipped on %s:\n\n\t%s\n", self.runbook.cloud, '\n\t'.join(skip_test_suites))
-        log.info("")
-        log.info("******** Waagent: Test Environments *****")
-        log.info("")
-        log.info("Will execute tests on %d environments:\n\n\t%s\n", len(environments), '\n\t'.join([env['c_env_name'] for env in environments]))
-        for env in environments:
-            log.info("Settings for %s:\n%s\n", env['c_env_name'], self._get_env_settings(env))
-        log.info("")
+            self._log.info("Skipping test suites %s", skip_test_suites)
 
         return environments
 
+    def create_existing_vm_environment(self) -> Dict[str, Any]:
+        loader = AgentTestLoader(self.runbook.test_suites, self.runbook.cloud)
+
+        vm: VirtualMachineClient = VirtualMachineClient(
+            cloud=self.runbook.cloud,
+            location=self.runbook.location,
+            subscription=self.runbook.subscription_id,
+            resource_group=self.runbook.resource_group_name,
+            name=self.runbook.vm_name)
+
+        ip_address = vm.get_ip_address()
+
+        return {
+            "c_env_name": self.runbook.vm_name,
+            "c_platform": [
+                {
+                    "type": "ready"
```

Comment:
"Ready" can be used to pass existing machine to LISA, "ready" meaning that they can be used without LISA initializing them or performing any operations on them. They are a good fit for existing VMs/VMSSs or to indicate to the AgentTestSuite that it needs to create a VMSS